Makefile

```diff
@@ -130,3 +137,27 @@ cargo-deny:
 	cargo install --version 0.9.1 cargo-deny --locked
 	cargo fetch
 	cargo deny --all-features --no-default-features check --disable-fetch licenses sources
+
+# Define a target to tag all images
+tag-images: $(TAG_IMAGES)
+
+# This defines the TAG_IMAGE variable, extracting the image name from the target name
+$(TAG_IMAGES): TAG_IMAGE = $(@:tag-%=%)
+
+$(TAG_IMAGES): check-publish-version
+	docker tag $(TAG_IMAGE) $(if $(PUBLISH_IMAGES_REPO), $(PUBLISH_IMAGES_REPO)/)$(TAG_IMAGE):$(PUBLISH_IMAGES_VERSION)
+
+# Define a target to tag all images
+publish-images: $(PUSH_IMAGES)
+
+# This defines the TAG_IMAGE variable, extracting the image name from the target name
+$(PUSH_IMAGES): TAG_IMAGE = $(@:push-%=%)
+
+
+$(PUSH_IMAGES): check-publish-version
+	docker push $(if $(PUBLISH_IMAGES_REPO), $(PUBLISH_IMAGES_REPO)/)$(TAG_IMAGE):$(PUBLISH_IMAGES_VERSION)
```

Comment:
> For pushing images, I think it makes sense to make PUBLISH_IMAGES_REPO required. With the current approach, the images will be pushed to the docker hub (provided that the user has logged in) since that is the default registry for docker push.

Yeah that's my fault, I defined the requirements and figured it would fail if you accidentally tried to push to dockerhub unless you had created all of those repos there. But I think you're right, it would be better to require the registry explicitly.

Also @ecpullen I think we should change it to `PUBLISH_IMAGES_REGISTRY`. I had the terminology wrong when we chatted about this. the URI parts are `registry/repo:version`